### PR TITLE
update xcat inventory osimage export and import cases

### PR DESCRIPTION
**Pull request by tingtli**
update xcat inventory osimage export and import cases

### CODE REVIEW

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 1]
```diff
@@ -930,20 +931,20 @@ check:rc==0
 cmd:chdef -t osimage -o test_myimage imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist
 check:rc==0
 cmd:if [ -e /opt/inventory/site/osimage/ ]; then cp -f /opt/inventory/site/osimage/ /opt/inventory/site/osimage.bak; else mkdir -p /opt/inventory/site/osimage/; fi
-cmd:xcat-inventory export -t osimage -o test_myimage --format yaml -d //opt/inventory/site/osimage/
+cmd:xcat-inventory export -t osimage -o test_myimage --format yaml -d /opt/inventory/site/osimage/
 check:rc==0
 check:output=~The osimage objects has been exported to directory /opt/inventory/site/osimage/
-cmd:otherpkglist=`lsdef -t osimage -o test_myimage |grep otherpkglist|awk -F= '{print $2}'`;diff $otherpkglist /opt/inventory/site/osimage/test_myimage/$otherpkglist
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: hi @tingtli , the `partitionfile` should also be checked

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 2]
```diff
@@ -914,3 +914,184 @@ check:rc==0
...
+#check:rc==0
+cmd:chdef -t osimage -o test_myimage imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist
+check:rc==0
+cmd:if [ -e /opt/inventory/site/osimage/ ]; then cp -f /opt/inventory/site/osimage/ /opt/inventory/site/osimage.bak; else mkdir -p /opt/inventory/site/osimage/; fi
+cmd:xcat-inventory export -t osimage -o test_myimage --format yaml -d /opt/inventory/site/osimage/
+check:rc==0
+check:output=~The osimage objects has been exported to directory /opt/inventory/site/osimage/
+cmd:otherpkglist=`lsdef -t osimage -o test_myimage |grep otherpkglist|awk -F= '{print $2}'`;diff $otherpkglist /opt/inventory/site/osimage/test_myimage$otherpkglist
+check:rc==0
+cmd:synclists=`lsdef -t osimage -o test_myimage |grep synclists|awk -F= '{print $2}'`;diff $synclists /opt/inventory/site/osimage/test_myimage$synclists
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: hi @tingtli , `partitionfile` should also be checked

**Comment by whowutwut** [comment]
@tingtli Can you execute this and show the output of the testing cases?  Are temp files cleaned up after the run, is anything left around?

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 3]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi
+check:rc==0
+#cmd:ssh $$DSTMN 'lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi'
+#check:rc==0
+cmd:chdef -t osimage -o test_myimage imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist partitionfile=/tmp/partitionfile
+check:rc==0
+cmd:if [ -e /opt/inventory/site/osimage ]; then cp -f /opt/inventory/site/osimage /opt/inventory/site/osimage.bak; else mkdir -p /opt/inventory/site/osimage; fi
+cmd:xcat-inventory export -t osimage -o test_myimage --format yaml -d /opt/inventory/site/osimage
+check:rc==0
+check:output=~The osimage objects has been exported to directory /opt/inventory/site/osimage
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: It is better to add below line to debug conveniently
```
cmd: ls -lFR /opt/inventory/site/osimage
```

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 4]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:echo "test" >> /tmp/exlist
+cmd:if [ -e /tmp/partitionfile ]; then cp -f /tmp/partitionfile /tmp/partitionfile.bak; fi
+cmd:echo "test" >> /tmp/partitionfile
+cmd:lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi
+check:rc==0
+#cmd:ssh $$DSTMN 'lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi'
+#check:rc==0
+cmd:chdef -t osimage -o test_myimage imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist partitionfile=/tmp/partitionfile
+check:rc==0
+cmd:if [ -e /opt/inventory/site/osimage ]; then cp -f /opt/inventory/site/osimage /opt/inventory/site/osimage.bak; else mkdir -p /opt/inventory/site/osimage; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: Line 935 won't work as expected when `` /opt/inventory/site/osimage `` have already existed.
It should be 
```
cmd:dir="/opt/inventory/site/osimage";if [ -e "${dir}" ];then mv ${dir} ${dir}".bak"; fi; mkdir -p $dir
```

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 5]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+check:rc==0
+cmd:synclists=`lsdef -t osimage -o test_myimage |grep synclists|awk -F= '{print $2}'`;diff $synclists /opt/inventory/site/osimage/test_myimage$synclists
+check:rc==0
+cmd:postinstall=`lsdef -t osimage -o test_myimage |grep postinstall|awk -F= '{print $2}'`;diff $postinstall /opt/inventory/site/osimage/test_myimage$postinstall
+check:rc==0
+cmd:exlist=`lsdef -t osimage -o test_myimage |grep exlist|awk -F= '{print $2}'`;diff $exlist /opt/inventory/site/osimage/test_myimage$exlist
+check:rc==0
+cmd:partitionfile=`lsdef -t osimage -o test_myimage |grep partitionfile|awk -F= '{print $2}'`;diff $partitionfile /opt/inventory/site/osimage/test_myimage$partitionfile
+check:rc==0
+cmd:pkglist=`lsdef -t osimage -o test_myimage |grep -w pkglist|awk -F= '{print $2}'`;if [ $pkglist == "/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist" ]; then echo 0; else echo 1; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: Line 971 won't work as design. 
It should be 
```
+cmd:pkglist=`lsdef -t osimage -o test_myimage |grep -w pkglist|awk -F= '{print $2}'`;if [ $pkglist == "/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist" ]; then exit 0; else exit 1; fi
```

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 6]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+check:rc==0
+cmd:template=`lsdef -t osimage -o test_myimage |grep template|awk -F= '{print $2}'`;if [ $template == "/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl" ]; then echo 0; else echo 1; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: It should be 
```
cmd:template=`lsdef -t osimage -o test_myimage |grep template|awk -F= '{print $2}'`;if [ $template == "/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl" ]; then exit 0; else exit 1; fi

```

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 7]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+check:rc==0
+cmd: rmdef -t osimage -o  test_myimage
+check:rc==0
+cmd: if [ -e /tmp/test_myimage.stanza ]; then cat /tmp/test_myimage.stanza |mkdef -z;fi
+cmd: if [ -e /opt/inventory/site/osimage/ ]; then rm -rf /opt/inventory/site/osimage/;fi
+cmd: if [ -e /opt/inventory/site/osimage.bak ]; then mv /opt/inventory/site/osimage.bak  /opt/inventory/site/osimage;fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: combine lien 978-979 to 
```
cmd:dir="/opt/inventory/site/osimage"; rm -rf $dir; if [ -d ${dir}".bak" ];then mv ${dir}".bak" $dir; fi
```

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 8]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:if [ -e /tmp/otherpkglist.bak ]; then mv -f /tmp/otherpkglist.bak /tmp/otherpkglist; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: change line 980 to 
```
cmd:file="/tmp/otherpkglist"; rm -rf $file; if [ -d ${file}".bak" ];then mv ${file}".bak" $file; fi

```

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 9]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:if [ -e /tmp/synclists.bak ]; then mv -f /tmp/synclists.bak /tmp/synclists; fi
+cmd:if [ -e /tmp/postinstall.bak ]; then mv -f /tmp/postinstall.bak /tmp/postinstall; fi
+cmd:if [ -e /tmp/exlist.bak ]; then mv -f /tmp/exlist.bak /tmp/exlist; fi
+cmd:if [ -e /tmp/partitionfile.bak ]; then mv -f /tmp/partitionfile.bak /tmp/partitionfile; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: line 981-984 have same problem with line 980

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 10]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:if [ -e /tmp/partitionfile ]; then cp -f /tmp/partitionfile /tmp/partitionfile.bak; fi
+cmd:echo "test" >> /tmp/partitionfile
+cmd:lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi
+check:rc==0
+#cmd:ssh $$DSTMN 'lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi'
+#check:rc==0
+cmd:chdef -t osimage -o test_myimage imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist partitionfile=/tmp/partitionfile
+check:rc==0
+cmd:if [ -e /opt/inventory/site/osimage ]; then cp -f /opt/inventory/site/osimage /opt/inventory/site/osimage.bak; else mkdir -p /opt/inventory/site/osimage; fi
+cmd:xcat-inventory export -t osimage -o test_myimage --format yaml -d /opt/inventory/site/osimage
```
Comment (whowutwut) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: What happens if this command is not installed,  should we test that in the beginning and just leave this test case bundle?

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 11]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+`check:rc==0
+cmd:lsdef -t osimage -o test_myimage2 >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage2 -z >/tmp/test_myimage2.stanza ;rmdef -t osimage -o test_myimage2;fi
+check:rc==0
+cmd:chdef -t osimage -o test_myimage1 imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist partitionfile=/tmp/partitionfile
+check:rc==0
+cmd:chdef -t osimage -o test_myimage2 imagetype=linux provmethod=install pkglist=/tmp/pkglist template=/tmp/template otherpkglist=/opt/xcat/share/xcat/install/rh/otherpkglist synclists=/opt/xcat/share/xcat/install/rh/synclists postinstall=/opt/xcat/share/xcat/install/rh/postinstall exlist=/opt/xcat/share/xcat/install/rh/exlist partitionfile=/opt/xcat/share/xcat/install/rh/partitionfile
+check:rc==0
+cmd:if [ -e /opt/inventory/site/osimage/ ]; then cp -f /opt/inventory/site/osimage /opt/inventory/site/osimage.bak; else mkdir -p /opt/inventory/site/osimage/; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: Same problem like 935

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 12]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:xcat-inventory export -t osimage -o test_myimage1,test_myimage2 --format yaml -d /opt/inventory/site/osimage
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: Maybe we can try ``--format json`` this time

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 13]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+check:rc==0
+cmd: if [ -e /tmp/test_myimage.stanza ]; then cat /tmp/test_myimage.stanza |mkdef -z;fi
+cmd: if [ -e /opt/inventory/site/osimage/ ]; then rm -rf /opt/inventory/site/osimage/;fi
+cmd: if [ -e /opt/inventory/site/osimage.bak ]; then mv /opt/inventory/site/osimage.bak  /opt/inventory/site/osimage;fi
+cmd:if [ -e /tmp/otherpkglist.bak ]; then mv -f /tmp/otherpkglist.bak /tmp/otherpkglist; fi
+cmd:if [ -e /tmp/synclists.bak ]; then mv -f /tmp/synclists.bak /tmp/synclists; fi
+cmd:if [ -e /tmp/postinstall.bak ]; then mv -f /tmp/postinstall.bak /tmp/postinstall; fi
+cmd:if [ -e /tmp/exlist.bak ]; then mv -f /tmp/exlist.bak /tmp/exlist; fi
+cmd:if [ -e /tmp/partitionfile.bak ]; then mv -f /tmp/partitionfile.bak /tmp/partitionfile; fi
+end
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: Below 2 cases have the same problems with above case

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 14]
```diff
@@ -914,3 +914,340 @@ check:rc==0
...
+cmd:if [ -e /tmp/pkglist ]; then cp -f /tmp/pkglist /tmp/pkglist.bak; fi
+cmd:echo "test" >> /tmp/pkglist
+cmd:if [ -e /tmp/template ]; then cp -f /tmp/template /tmp/template.bak; fi
+cmd:echo "test" >> /tmp/template
+cmd:if [ -e /opt/xcat/share/xcat/install/rh/otherpkglist ]; then cp -f /opt/xcat/share/xcat/install/rh/otherpkglist /opt/xcat/share/xcat/install/rh/otherpkglist.bak; fi
+cmd:echo "test" >> /opt/xcat/share/xcat/install/rh/otherpkglist
+cmd:if [ -e /opt/xcat/share/xcat/install/rh/synclists ]; then cp -f /opt/xcat/share/xcat/install/rh/synclists /opt/xcat/share/xcat/install/rh/synclists.bak; fi
+cmd:echo "test" >> /opt/xcat/share/xcat/install/rh/synclists
+cmd:if [ -e /opt/xcat/share/xcat/install/rh/synclists ]; then cp -f /opt/xcat/share/xcat/install/rh/synclists /opt/xcat/share/xcat/install/rh/synclists.bak; fi
+cmd:echo "test" >> /opt/xcat/share/xcat/install/rh/synclists
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: Is this valid to change file under ``/opt/xcat/share/xcat/install/rh``? that means original osimage has ``test``, but after export and import, the ``test`` will all gone?

**Comment by tingtli** [comment]
@ whowutwut  Yes. I have tested all the cases. Attached are the result files. And I removed all the temporary files.

**Comment by whowutwut** [comment]
@tingtli  I don't see the test results, did you forgot to attach it?

**Comment by tingtli** [comment]
@whowutwut  Please check again.

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 15]
```diff
@@ -1065,53 +1068,53 @@ check:output=~The object test_myimage1 has been imported
...
 check:rc==0
-cmd:postinstall=`lsdef -t osimage -o test_myimage2 |grep postinstall|awk -F= '{print $2}'`;if [ $postinstall == "/opt/xcat/share/xcat/install/rh/postinstall" ]; then echo 0; else echo 1; fi
+cmd:postinstall=`lsdef -t osimage -o test_myimage2 |grep postinstall|awk -F= '{print $2}'`;if [ $postinstall == "/opt/xcat/share/xcat/install/rh/postinstall" ]; then exit 0; else exit 1; fi
 check:rc==0
-cmd:exlist=`lsdef -t osimage -o test_myimage2 |grep exlist|awk -F= '{print $2}'`;if [ $exlist == "/opt/xcat/share/xcat/install/rh/exlist" ]; then echo 0; else echo 1; fi
+cmd:exlist=`lsdef -t osimage -o test_myimage2 |grep exlist|awk -F= '{print $2}'`;if [ $exlist == "/opt/xcat/share/xcat/install/rh/exlist" ]; then exit 0; else exit 1; fi
 check:rc==0
-cmd:partitionfile=`lsdef -t osimage -o test_myimage2 |grep partitionfile|awk -F= '{print $2}'`;if [ $partitionfile == "/opt/xcat/share/xcat/install/rh/partitionfile" ]; then echo 0; else echo 1; fi
+cmd:partitionfile=`lsdef -t osimage -o test_myimage2 |grep partitionfile|awk -F= '{print $2}'`;if [ $partitionfile == "/opt/xcat/share/xcat/install/rh/partitionfile" ]; then exit 0; else echo 1; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: This line still has ``echo 1``

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 16]
```diff
@@ -914,3 +914,344 @@ check:rc==0
...
+cmd:if [ -e /tmp/postinstall ]; then cp -f /tmp/postinstall /tmp/postinstall.bak; fi
+cmd:echo "test" >> /tmp/postinstall
+cmd:if [ -e /tmp/exlist ]; then cp -f /tmp/exlist /tmp/exlist.bak; fi
+cmd:echo "test" >> /tmp/exlist
+cmd:if [ -e /tmp/partitionfile ]; then cp -f /tmp/partitionfile /tmp/partitionfile.bak; fi
+cmd:echo "test" >> /tmp/partitionfile
+cmd:lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi
+check:rc==0
+#cmd:ssh $$DSTMN 'lsdef -t osimage -o test_myimage >/dev/null 2>&1;if [[ $? -eq 0 ]]; then lsdef -t osimage -o test_myimage -z >/tmp/test_myimage.stanza ;rmdef -t osimage -o test_myimage;fi'
+#check:rc==0
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: It is better to remove lines 931-932 if they are useless any more.

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 17]
```diff
@@ -914,3 +914,344 @@ check:rc==0
...
+check:rc==0
+cmd:otherpkglist=`lsdef -t osimage -o test_myimage2 |grep otherpkglist|awk -F= '{print $2}'`;if [ $otherpkglist == "/opt/xcat/share/xcat/install/rh/otherpkglist" ]; then exit 0; else exit 1; fi
+check:rc==0
+cmd:synclists=`lsdef -t osimage -o test_myimage2 |grep synclists|awk -F= '{print $2}'`;if [ $synclists == "/opt/xcat/share/xcat/install/rh/synclists" ]; then exit 0; else exit 1; fi
+check:rc==0
+cmd:postinstall=`lsdef -t osimage -o test_myimage2 |grep postinstall|awk -F= '{print $2}'`;if [ $postinstall == "/opt/xcat/share/xcat/install/rh/postinstall" ]; then exit 0; else exit 1; fi
+check:rc==0
+cmd:exlist=`lsdef -t osimage -o test_myimage2 |grep exlist|awk -F= '{print $2}'`;if [ $exlist == "/opt/xcat/share/xcat/install/rh/exlist" ]; then exit 0; else exit 1; fi
+check:rc==0
+cmd:partitionfile=`lsdef -t osimage -o test_myimage2 |grep partitionfile|awk -F= '{print $2}'`;if [ $partitionfile == "/opt/xcat/share/xcat/install/rh/partitionfile" ]; then exit 0; else echo 1; fi
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: This line still has ``echo 1``

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 18]
```diff
@@ -914,3 +914,344 @@ check:rc==0
...
+cmd:echo "test" >> /opt/xcat/share/xcat/install/rh/template
+cmd:chdef -t osimage -o test_myimage1 imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist partitionfile=/tmp/partitionfile
+check:rc==0
+cmd:chdef -t osimage -o test_myimage2 imagetype=linux provmethod=install pkglist=/tmp/pkglist template=/tmp/template otherpkglist=/opt/xcat/share/xcat/install/rh/otherpkglist synclists=/opt/xcat/share/xcat/install/rh/synclists postinstall=/opt/xcat/share/xcat/install/rh/postinstall exlist=/opt/xcat/share/xcat/install/rh/exlist partitionfile=/opt/xcat/share/xcat/install/rh/partitionfile
+check:rc==0
+cmd:dir="/opt/inventory/site/osimage";if [ -e "${dir}" ];then mv ${dir} ${dir}".bak"; fi; mkdir -p $dir
+cmd:xcat-inventory export -d /opt/inventory/site/
+check:rc==0
+check:output=~The osimage objects has been exported to directory /opt/inventory/site/
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: It is better to add ``cmd: ls -lFR  /opt/inventory/site/``

File: xCAT-test/autotest/testcase/xcat-inventory/cases.osimage [hunk 19]
```diff
@@ -914,3 +914,344 @@ check:rc==0
...
+cmd:if [ -e /opt/xcat/share/xcat/install/rh/pkglist ]; then cp -f /opt/xcat/share/xcat/install/rh/pkglist /opt/xcat/share/xcat/install/rh/pkglist.bak; fi
+cmd:echo "test" >> /opt/xcat/share/xcat/install/rh/pkglist
+cmd:if [ -e /opt/xcat/share/xcat/install/rh/template ]; then cp -f /opt/xcat/share/xcat/install/rh/template /opt/xcat/share/xcat/install/rh/template.bak; fi
+cmd:echo "test" >> /opt/xcat/share/xcat/install/rh/template
+cmd:chdef -t osimage -o test_myimage1 imagetype=linux provmethod=install pkglist=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl otherpkglist=/tmp/otherpkglist synclists=/tmp/synclists postinstall=/tmp/postinstall exlist=/tmp/exlist partitionfile=/tmp/partitionfile
+check:rc==0
+cmd:chdef -t osimage -o test_myimage2 imagetype=linux provmethod=install pkglist=/tmp/pkglist template=/tmp/template otherpkglist=/opt/xcat/share/xcat/install/rh/otherpkglist synclists=/opt/xcat/share/xcat/install/rh/synclists postinstall=/opt/xcat/share/xcat/install/rh/postinstall exlist=/opt/xcat/share/xcat/install/rh/exlist partitionfile=/opt/xcat/share/xcat/install/rh/partitionfile
+check:rc==0
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.osimage]: It is better to add ``lsdef test_myimage1,test_myimage2 ``

**Comment by hu-weihua** [comment]
It is ok for me. thanks @tingtli

**Comment by whowutwut** [comment]
I'm still trying to understand what is going on here.. Comments:

*  The `/tmp/osimages` name is not great.  We should create unique directories so we can quickly identify that it was created by xCAT-test,  example:  `/tmp/xcattest_osimages` 
*  What's the need to copy all the files to `.bak`? 
    If it's to debug when errors occur, then leave the `/tmp/xcattest_osimages` directory around, then on the next run of the testcase, `rm -rf /tmp/xcattest_osimages` and start clean.  
* This is really bad: 
   `cp -f /opt/xcat/share/xcat/install/rh/synclists /opt/xcat/share/xcat/install/rh/synclists.bak;` 

   Essentially we are duplicating files onto the xCAT install if a user runs xCAT-test, those files are NOT tracked by RPM and would never get cleaned up correctly. 

* Why are not variables being used? What if I wanted to ask the osimage name to be changed to `test_myimage` -> `xcattest_inventory_case` ... 

* If you are creating two osimages, `test_myimage1` and `test_myimage2` , both are using `chdef` command to create it.  You should change one to using `mkdef` so that we have coverage on both case.  

* Why create the directory here: `/opt/inventory/site/osimage` ?? 
  Temporary working directories should be in /tmp, don't add new files into the filesystem that cannot be managed by rpms... 

```
RUN:xcat-inventory export -d /opt/inventory/site/ [Sat Jan 15 16:25:29 2000]
ElapsedTime:2 sec
RETURN rc = 2
OUTPUT:
usage: xcat-inventory [-h] [--debug] [-v] [-V] <subcommand> ...
xcat-inventory: error: unrecognized arguments: -d /opt/inventory/site/
CHECK:rc == 0	[Failed]
```

This above case is dependent on function added onto a certain version of xcat-inventory,  Any check for this? Or maybe the test case should check the man page and only run if that option is there. 

These test cases are not reporting correct error: 
```
RUN:otherpkglist=`lsdef -t osimage -o test_myimage1 |grep otherpkglist|awk -F= '{print $2}'`;diff -y $otherpkglist /opt/inventory/site/osimage/test_myimage1$otherpkglist [Sat Jan 15 16:25:31 2000]
ElapsedTime:0 sec
RETURN rc = 2
OUTPUT:
diff: /opt/inventory/site/osimage/test_myimage1/tmp/otherpkglist: No such file or directory

```

Possibly because of the issue from above where there's no `-d` option here...

**Pull request merged**